content/docs/concepts/traffic-management/routing-rules/index.md

```diff
@@ -0,0 +1,271 @@
+destination workloads. Additionally, these hosts do not even necessarily
+correspond to an actual routable service in the mesh.
+
+For example, to configure routing rules for requests to the `reviews` service
+using its internal mesh name `reviews` or via its `bookinfo.com` host, you can
+configure a virtual service with the `hosts:` key as follows:
+
+{{< text yaml >}}
+hosts:
+  - reviews
+  - bookinfo.com
+{{< /text >}}
+
+The `hosts:` key specifies, implicitly or explicitly, one or more fully
+qualified domain names (FQDN).
+
+The short name `reviews`, can implicitly expand to an implementation specific
+FQDN. In a Kubernetes environment, the full name is derived from the cluster
+and namespace of the virtual service, for example,
+`reviews.default.svc.cluster.local`.
+
+## Precedence
+
+When there are multiple rules for a given destination, they are evaluated in
+the order they appear in the virtual service. The first rule on the list has
+the highest priority.
+
+### Why is precedence important?
+
+Whenever the routing story of a particular service is purely weight based, you can specify it in a single rule.
+
+When you use other conditions to route traffic, such as requests from a
+specific user, you need more than one rule to specify the routing. These cases
+require careful consideration of priorities to ensure **Pilot** evaluates the
+rules in the right order.
+
+A common pattern for generalized route specification is:
+
+1. Provide one or more higher priority rules that match the various conditions.
+
+1. Provide a single weight-based rule with no match condition last to provide
+   the weighted distribution of traffic for all other cases.
+
+The following virtual service contains two rules to specify that **Pilot**
+sends all requests for the `reviews` service including the `Foo` header with
+the `bar` value to the instances in the `v2` subset and send all remaining
+requests to the instances in the `v1` subset:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: reviews
+spec:
+  hosts:
+  - reviews
+  http:
+  - match:
+    - headers:
+        Foo:
+          exact: bar
+    route:
+    - destination:
+        host: reviews
+        subset: v2
+  - route:
+    - destination:
+        host: reviews
+        subset: v1
+{{< /text >}}
+
+The header-based rule has the higher priority because it comes first in the
+configuration file. If the header-based rule was lower, these rules wouldn't
+work as expected. **Pilot** would evaluate the weight-based rule first in that
+case and route all traffic to the instances in the `v1` subset, even requests
+including the matching `Foo` header.
+
+Once **Pilot** finds a rule that applies to the incoming request, the rule is
+executed and the rule-evaluation process terminates.
+
+{{< warning_icon >}} Consider the priorities of each rule in a configuration file when there is more than one.
+
+## Multiple match conditions {#multi-condition}
```

Comment:
Multiple match conditions are "conditional rules", so this section belongs after the following one (possibly even as a subsection of it).

Reply:
Agreed, I'll move the content then.